src/figuresClasses.ts

```diff
@@ -1,11 +1,101 @@
-export interface Figure {}
+type ColorType = 'red' | 'green' | 'blue';
+type ShapeType = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+const PI = Math.PI;
 
-export class Circle implements Figure {}
+export interface Figure {
+  shape: ShapeType;
+  color: ColorType;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  color: ColorType;
+
+  shape: ShapeType = 'triangle';
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: ColorType, a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side lengths must be greater than zero.');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error(
+        // eslint-disable-next-line max-len
+        'The longest side of a triangle must be less than the sum of the other two.',
+      );
+    }
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
```

Comment:
When declaring a constructor in TypeScript, if you use an access modifier on a parameter (e.g., `public` or `private`), TypeScript will automatically create and initialize a property with that name. You don't need to assign parameter values to properties with the same name explicitly.